pybind11: 3.0.0

ax3l opened the pull request:
Update to pybind11 v3.0.0
https://github.com/pybind/pybind11/releases/tag/v3.0.0

ax3l committed:
pybind11: 3.0.0

Update to pybind11 v3.0.0
https://github.com/pybind/pybind11/releases/tag/v3.0.0

roelof-groenewald reviewed (approved):
Is this update to fix the issue with `PyCapsule`?

ax3l commented:
Yes, this will add typehints for capsule :)
- https://github.com/sizmailov/pybind11-stubgen/pull/258#issuecomment-3127850700
- https://github.com/pybind/pybind11/releases/tag/v3.0.0

I will add one more commit to modernize enum types to a new best-practice.

We then should update WarpX and ImpactX to pybind11 requirements as well with the next weekly update today.

ax3l committed:
Update Enum Types

https://pybind11.readthedocs.io/en/stable/classes.html#enumerations-and-internal-types

ax3l commented:
Will push one more time, so that tests pass...
